src/schema/v2/artist/index.ts

```diff
@@ -399,6 +399,13 @@ export const ArtistType = new GraphQLObjectType<any, ResolverContext>({
         },
       },
       carousel: ArtistCarousel,
+      artworksCountWithinCollection: {
```

Comment:
I think this is fine, but an interesting possible alternative schema design (which I think does fit our data a bit more accurately), is to have this data live on the _edge_ of the artist connection present on a Collection.

This avoids needing to add a field to `Artist` which depends on the structure of the query (that's overhead that if we do a lot of that kind of thing gets unwieldy).

Also, putting metadata on the edge from a Collection to an artist node, describing something specific about _that_ relationship (such as `createdAt` -> when the artist was added to the collection, `artworksCount`, and so forth) is a good pattern (and one which we do already in Metaphysics), IMO.

The Gravity update to expose this field in JSON is correct (it needs to be exposed), the schema change would just be in Metaphysics, to add this count to the edge field.

Reply:
That makes a lot of sense! We were wondering where we could expose these context-specific fields, and this is a great solution.